escodegen.js

```diff
@@ -1943,10 +1966,14 @@
                         result.push(' ');
                     }
                 }
-                result.push('.');
+                result.push(expr.optional ? '?.' : '.');
```

Comment:
The additional ` ` added by the preceding block is unnecessary when using optional chaining - `1?.a` works fine. This looks like it will codegen that as `1 ?.a`.

Reply:
Nice catch!

Reply:
https://github.com/estools/escodegen/pull/412/commits/ba05f2050732d9f9504237f708e180ba9320d4b2